src/library/basetrackcache.cpp

```diff
@@ -145,18 +146,59 @@ void BaseTrackCache::setSearchColumns(const QStringList& columns) {
 }
 
 TrackPointer BaseTrackCache::lookupCachedTrack(TrackId trackId) const {
-    TrackPointer pTrack;
-    if (m_bIsCaching) {
-        pTrack = TrackCache::instance().lookupById(trackId).getTrack();
-        // After obtaining a strong reference of the Track object the lock
-        // on TrackCache has been released instantly to reduce lock contention!
-        if (pTrack && pTrack->isDirty()) {
-            m_dirtyTracks.insert(trackId);
+    DEBUG_ASSERT(m_bIsCaching);
+    if (m_cachedTrackId != trackId) {
+        refreshCachedTrack(std::move(trackId));
+    }
+    return m_cachedTrackPtr;
+}
+
+void BaseTrackCache::refreshCachedTrack(TrackId trackId) const {
+    DEBUG_ASSERT(m_bIsCaching);
+    if (trackId.isValid()) {
+        refreshCachedTrack(
+                std::move(trackId),
```

Comment:
Here is looks like trackId is used after move(). I the compiler does it probably right here, but is it guaranteed?